commitment/taproot.go

```diff
@@ -243,28 +293,17 @@ func TapBranchHash(preimage []byte) (*chainhash.Hash, error) {
 // TapLeafHash computes the TapHash of a TapLeaf node from its preimage
 // if possible, otherwise an error is returned.
 func TapLeafHash(preimage []byte) (*chainhash.Hash, error) {
-	// Empty preimage.
-	if len(preimage) == 0 {
-		return nil, ErrInvalidEmptyTapscriptPreimage
+	// Decode the script version and the script itself.
+	version, script, err := ScriptFromLeafPreimage(preimage)
```

Comment:
Not quite correct here in how this is used. Please rename the function from "TapLeafHash" to something else, as I think it's confusing in how it's used, which seems to be closer to "ProofNotTapLeafHashCommitsToTapleafAssetCommitment" therefore an empty preimage is valid here

Reply:
Empty preimage would be rejected inside `ScriptFromLeafPreimage`?

Agree re: naming, could lift this validation to `TapscriptPreimage.TapHash()`.

Reply:
Made a suggestion above, then I think the name of this could be like:
```
func ExtractTapLeafscript(t TapscriptLeaf) (uint32, Tapscript, error) {
}
```

or w/e 

Or `NewTapscriptFromLeaf`. 

Then similar to this, we'd have a constructor to create `TapscriptLeaf` (maybe force a private var?) that asserts that the version and prefix are in place.